odl/solvers/util/callback.py

```diff
@@ -325,12 +328,18 @@ def __init__(self, *args, **kwargs):
         self.kwargs = kwargs
         self.fig = kwargs.pop('fig', None)
         self.display_step = kwargs.pop('display_step', 1)
+        self.saveto = kwargs.pop('saveto', None)
         self.iter = 0
 
     def __call__(self, x):
         """Show the current iterate."""
         if (self.iter % self.display_step) == 0:
-            self.fig = x.show(*self.args, fig=self.fig, **self.kwargs)
+            if self.saveto is None:
+                self.fig = x.show(*self.args, fig=self.fig, **self.kwargs)
+            else:
+                self.fig = x.show(*self.args, fig=self.fig,
+                                  saveto=(self.saveto + '_' + str(self.iter)),
```

Comment:
Two comments:

- Add a format suffix. Right now it's storing in ??? format, which is whatever matplotlib defaults to. Perhaps use another kwarg to specify the format (suffix), defaulting to `'png'`.
- If possible, use an enumeration scheme that preserves order on disk. If you're unlucky, you get `iter_10` before `iter_2`. I'm not sure how to do it best, but at least test if this enumeration gives the right ordering.